libraries/ms-common/inc/blink_event_generator.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+// Takes an event ID with a data field and generates events toggling the data field between 1 and 0.
+// Requires interrupts, soft timers, and the event queue to be initialized.
+
+#include "event_queue.h"
+#include "soft_timer.h"
+
+typedef struct {
+  uint32_t interval_us;
+  uint16_t first_value;  // value of the first event; must be 1 or 0, default is 0 if not specified
+} BlinkEventGeneratorSettings;
+
+typedef struct {
+  uint32_t interval_us;
+  EventId event_id;
+  uint16_t first_value;
+  uint16_t next_value;
```

Comment:
change this to `current_state`, and change `first_value` to `default_state`, and typedef an enum called  `BlinkerState`, as in `BLINKER_STATE_ON`, and `BLINKER_STATE_OFF`. Right now the user can pass any `uint16_t` as the first value, which wouldn't really make sense... but if you typedef them as `BlinkerState` it's clear.